src/main/java/edu/hm/hafner/coverage/parser/JacocoParser.java

```diff
@@ -316,16 +316,4 @@ private void readValueCounter(final Node node, final StartElement startElement)
             }
         }
     }
-
-    private Value createValue(final String currentType, final int covered, final int missed) {
-        if (VALUE_COMPLEXITY.equals(currentType)) {
-            return new CyclomaticComplexity(covered + missed);
-        }
-        else {
-            var builder = new CoverageBuilder();
-            return builder.withMetric(Metric.valueOf(currentType))
-                        .withCovered(covered)
-                        .withMissed(missed).build();
-        }
-    }
```

Comment:
I think this should be private because it is used by JaCoCo and not intended to be pulled up.

Reply:
I moved that method to the base `CoverageParser` class and made it protected. As clover reports have the specific scenario of hitting `CONDITIONAL` blocks and converting those to `BRANCH` metric values, so `CloverParser` overrides `CoverageParser#createValue()`

Reply:
Update, I removed the need for `CoverageParser` to override the method for the `CONDITIONAL` to `BLOCK` scenario, but it stands that it needs to be `protected` so it can be called by the children parsers.